packages/odyssey-react-mui/src/Buttons/BaseMenuButton.tsx

```diff
@@ -174,31 +216,52 @@ const MenuButton = ({
         id={`${uniqueId}-button`}
         isDisabled={isDisabled}
         label={buttonLabel}
+        children={buttonChildren}
         onClick={openMenu}
         size={size}
         tooltipText={tooltipText}
         translate={translate}
         variant={buttonVariant}
       />
 
-      <MuiMenu
-        anchorOrigin={anchorOrigin}
-        transformOrigin={transformOrigin}
-        anchorEl={anchorEl}
-        id={`${uniqueId}-menu`}
-        MenuListProps={menuListProps}
-        onClose={closeMenu}
-        open={isOpen}
-      >
-        <MenuContext.Provider value={providerValue}>
-          {children}
-        </MenuContext.Provider>
-      </MuiMenu>
+      {children && (
+        <MuiMenu
+          anchorOrigin={anchorOrigin}
+          transformOrigin={transformOrigin}
+          anchorEl={anchorEl}
+          id={`${uniqueId}-menu`}
+          MenuListProps={menuListProps}
+          onClose={closeMenu}
+          open={isOpen}
+        >
+          <MenuContext.Provider value={providerValue}>
+            {children}
+          </MenuContext.Provider>
+        </MuiMenu>
+      )}
+
+      {popoverContent && (
+        <MuiPopover
+          open={isOpen}
+          anchorEl={anchorEl}
+          onClose={closeMenu}
+          anchorOrigin={anchorOrigin}
+          transformOrigin={transformOrigin}
+        >
+          <Box
+            sx={{
+              padding: odysseyDesignTokens.Spacing4,
+            }}
+          >
```

Comment:
We should make sure these styles are memoized.

Reply:
@KevinGhadyani-Okta this is the initial pass and has been closed. Please look at #2414 instead.